File: binary_sign_tool/api/src/sign_tool_service_impl.cpp
/*
 * Copyright (c) 2025-2025 Huawei Device Co., Ltd.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

#include "sign_tool_service_impl.h"
#include "pkcs7_data.h"
#include "signature_tools_errno.h"
#include "self_sign_sign_provider.h"
#include "local_sign_provider.h"
#include "signature_tools_log.h"
#include "param_constants.h"
#include "constant.h"
#include "remote_sign_provider.h"
#include "verify_elf.h"

namespace OHOS {
namespace SignatureTools {
bool SignToolServiceImpl::Sign(Options* options)
{
    std::string inFile = options->GetString(Options::IN_FILE);
    if (!FileUtils::isElfFile(inFile)) {
        SIGNATURE_TOOLS_LOGE("inFile is not a elf file");
        return false;
    }
    std::string mode = options->GetString(Options::MODE);
    std::string selfSign = options->GetString(Options::SELF_SIGN);
    std::shared_ptr<SignProvider> signProvider;
    if (ParamConstants::SELF_SIGN_TYPE_1 == selfSign) {
        signProvider = std::make_shared<SelfSignSignProvider>();
    } else if (LOCAL_SIGN == mode) {
        signProvider = std::make_shared<LocalSignProvider>();
    } else if (REMOTE_SIGN == mode) {
        signProvider = std::make_shared<RemoteSignProvider>();
    } else {
        SIGNATURE_TOOLS_LOGE("Resign mode. But not implemented yet");
        return false;
    }
    return signProvider->SignElf(options);
}

bool SignToolServiceImpl::Verify(Options* option)
{
    VerifyElf verifyElf;
    if (!verifyElf.Verify(option)) {
        return false;
    }
    return true;
}
} // namespace SignatureTools
} // namespace OHOS

File: binary_sign_tool/java/binary_sign_tool_lib/src/main/java/com/ohos/hapsigntool/api/SignToolServiceImpl.java
/*
 * Copyright (c) 2026-2026 Huawei Device Co., Ltd.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.ohos.hapsigntool.api;

import com.ohos.hapsigntool.entity.Options;
import com.ohos.hapsigntool.entity.ParamConstants;
import com.ohos.hapsigntool.hap.provider.LocalJKSSignProvider;
import com.ohos.hapsigntool.hap.provider.RemoteSignProvider;
import com.ohos.hapsigntool.hap.provider.SelfSignSignProvider;
import com.ohos.hapsigntool.hap.provider.SignProvider;
import com.ohos.hapsigntool.utils.LogUtils;

import org.bouncycastle.jce.provider.BouncyCastleProvider;

import java.security.Security;

/**
 * Main entry of lib.
 *
 * @since 2021/12/28
 */
public class SignToolServiceImpl implements ServiceApi {
    /**
     * Logger.
     */
    private static final LogUtils LOGGER = new LogUtils(ServiceApi.class);

    static {
        Security.addProvider(new BouncyCastleProvider());
    }

    /**
     * Sign for hap/elf files.
     *
     * @param options options
     * @return Sign or not
     */
    @Override
    public boolean signHap(Options options) {
        String mode = options.getString(Options.MODE, Options.LOCAL_SIGN);
        String selfSign = options.getString(Options.SELF_SIGN);

        // Sign provider selection
        SignProvider signProvider;
        if (ParamConstants.SELF_SIGN_TYPE_1.equals(selfSign)) {
            // Self-sign mode
            signProvider = new SelfSignSignProvider();
            LOGGER.info("Using SelfSignSignProvider");
        } else if ("localSign".equalsIgnoreCase(mode)) {
            // Local sign mode
            signProvider = new LocalJKSSignProvider();
            LOGGER.info("Using LocalJKSSignProvider");
        } else if ("remoteSign".equalsIgnoreCase(mode)) {
            // Remote sign mode
            signProvider = new RemoteSignProvider();
            LOGGER.info("Using RemoteSignProvider");
        } else {
            LOGGER.error("Unsupported mode: {}", mode);
            return false;
        }

        // Sign ELF files
        return signProvider.signElf(options);
    }
}
